README.md

```diff
@@ -1,5 +1,40 @@
-# lighthouse 
-Controller to facilitate DNS discovery in multi-cluster connected environments (proof-of-concept state)
+# Lighthouse 
+Lighthouse provides DNS discovery to clusters connected by [Submariner](https://github.com/submariner-io/submariner) in multi-cluster connected environments. The solution aims to compatible with any CNI-compatible cluster network provider.
+
+# Architecture
+The below digram shows the basic Lighthouse architecture.
+
+![Lighthouse Architecture](https://raw.githubusercontent.com/submariner-io/lighthouse/master/docs/img/architecture.png)
+
+## Lighthouse Controller
+This is the central discovery controller that gathers the information from the clusters, decides what information is to be shared and distributes the information as newly defined CRDs.
+
+The Lighthouse controller will be running along with the [Admiral](https://github.com/submariner-io/admiral) control plane. It uses the API’s exposed by the Admiral to connect to each cluster and listen for Service CRUD. Once it is notified that a new Service is created and if the Service needs to be distributed, it frames the Lighthouse CRD and the same is distributed to all the clusters that are part of the control plane. The flow for update and delete will be similar.
+
+### WorkFlow
+The typical workflow is as follows.
+
+- Lighthouse controller registers with Admiral for cluster join and unjoin events.
+- Once it is notified about a join event, it retrieves the credentials from Admiral and registers for a call back on Service creation with a shared informer.
+- When a new Service is created in a cluster, the Lighthouse controller will be notified about it and it will create the Lighthouse CRD with the Service info.
+- The controller with the help of Admiral distributes the CRD to all the clusters that have joined.
+
+![Lighthouse Controller WorkFlow](https://raw.githubusercontent.com/submariner-io/lighthouse/master/docs/img/controllerWorkFlow.png)
+
+## Lighthouse Plugin
+The Lighthouse CoreDNS plugin will work along with the default Kubernetes plugin. It will use the Lighthouse CRD that is distributed by the controller for DNS resolution. The below diagram indicates a high-level architecture.
+
+![Lighthouse Plugin Architecture](https://raw.githubusercontent.com/submariner-io/lighthouse/master/docs/img/lighthousePluginArchitecture.png)
+
+### WorkFlow
+The typical workflow is as follows.
+
+- A pod tries to resolve a Service name.
+- The Kubernetes plugin in CoreDNS will try to resolve the request if no records are present the request will be sent to Lighthouse plugin
+- The Lighthouse plugin will try to resolve the request using its cache that is built using the CRD distribute by Lighthouse controller.
+- If a record exists it will be returned or else the plugin will pass the request to the next plugin registered in kube-dns.
```

Comment:
- "- If a record exists it will be returned, otherwise the plugin will pass the request to the next plugin registered in kube-dns."

Reply:
Done.